aptos-move/aptos-aggregator/src/aggregator_change_set.rs

```diff
@@ -2,331 +2,388 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    bounded_math::{code_invariant_error, expect_ok, BoundedMath, SignedU128},
-    delta_math::{merge_data_and_delta, merge_two_deltas, DeltaHistory},
-    types::{AggregatorID, SnapshotToStringFormula},
+    bounded_math::code_invariant_error,
+    delta_change_set::DeltaOp,
+    types::{AggregatorID, AggregatorValue, SnapshotToStringFormula},
 };
 use move_binary_format::errors::PartialVMResult;
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub enum AggregatorChange {
-    Data {
-        value: u128,
-    },
-    StringData {
-        value: Vec<u8>,
-    },
+pub enum AggregatorApplyChange {
     AggregatorDelta {
-        delta: SignedU128,
-        max_value: u128,
-        history: DeltaHistory,
+        delta: DeltaOp,
     },
-    /// Value of delta is:
-    /// formula(value of base_aggregator at the beginning of the transaction + delta)
+    /// Value is:
+    /// (value of base_aggregator at the beginning of the transaction + delta)
     SnapshotDelta {
-        delta: SignedU128,
         base_aggregator: AggregatorID,
+        delta: DeltaOp,
+    },
+    /// Value is:
+    /// formula(value of base_snapshot at the end of the transaction)
```

Comment:
Why is it the value of base_snapshot at the end of the transaction? It should be the value at the point where the snapshot is created right?

Reply:
Ah... I now get it. This AggregatorID points to a snapshot which doesn't change after it is created. Confused because we are using AggregatorID for both aggregators and snapshots.